util/libcrypto.num

```diff
@@ -4741,7 +4741,7 @@ OSSL_CRMF_CERTTEMPLATE_get0_serialNumber 4688	3_0_0	EXIST::FUNCTION:CMP
 OSSL_CRMF_CERTTEMPLATE_get0_issuer      4689	3_0_0	EXIST::FUNCTION:CMP
 OSSL_CRMF_CERTTEMPLATE_fill             4690	3_0_0	EXIST::FUNCTION:CMP
 OSSL_CRMF_ENCRYPTEDVALUE_get1_encCert   4691	3_0_0	EXIST::FUNCTION:CMP
-ERR_load_CRMF_strings                   4692	3_0_0	EXIST::FUNCTION:CMP
+ERR_load_CRMF_strings                   4692	3_0_0	EXIST::FUNCTION:
```

Comment:
I wonder why this removes `:CMP` ?

Reply:
I was wondering, too.
I fear that this is due to a shortcoming of `util/mkerr.pl`, which for some reason does not add 
```
# ifndef OPENSSL_NO_CMP
```
to the generated file `include/openssl/crmferr.h` (which includes the declaration of `ERR_load_CRMF_strings()`), although `Configure` contains:
```
        # no-cmp also disables building of crmf
        $skipdir{"crypto/crmf"} = $what if $what eq "cmp";
```
which indicates to disable CRMF if CMP is disabled.

Nevertheless, the issue in `util/libcrypto.num` disappeared after following @mspncp´s change request to move our global improvements to `util/mkerr.pl` (and the adaptations to the files in generates) to a separate PR.

Reply:
So there is a pre-existing problem with the crmferr.h header file:
As soon as it gets regenerated using `make errors ERROR_REBUILD=-rebuild`
it does look differently, (not having the `#ifndef OPENSSL_NO_CMP` guard),
and once that happens `make ordinals` changes it's mind about
`ERR_load_CRMF_strings:CMP`

Reply:
Yes. So how to solve this? 
As mentioned, the issue is due to `util/mkerr.pl`.
Meanwhile I found the problem: it reads `@disablables` from `configdata.pm`, and this does not take into account the indirect dependency, stated in `Configure`, not to build CRMF if CMP is disabled :-(

Reply:
> not to build CRMF if CMP is disabled :-(

It seems like an entry is missing in `@disable_cascades`?

https://github.com/openssl/openssl/blob/4679345149f04eece835593823932263d9421456/Configure#L477-L489

Reply:
Maybe the correct entry in `@disable_cascades` would make this change superfluous, too?

https://github.com/openssl/openssl/blob/cb1e7e08bce2e20d8a9da0e418c2a8e58c7b055b/Configure#L1228-L1230

_(@levitte please correct me if I'm wrong.)_

Reply:
Looking a few lines up it seems that your change is indeed misplaced. The other fixups are just about correcting directory names.

https://github.com/openssl/openssl/blob/cb1e7e08bce2e20d8a9da0e418c2a8e58c7b055b/Configure#L1224-L1229

Reply:
Thanks @mspncp for pointing this out. Yet even when I replace in `Configure`
```
$skipdir{"crypto/crmf"} = $what if $what eq "cmp"; 
```
by
```
    "cmp"                    => [ "crmf" ],
```
at the end of the 
```
my @disable_cascades = (
```
declaration (which is certainly good for other uses of the dependency), it does not help correcting `include/openssl/crmferr.h`.
Which is not really a surprise since `util/mkerr.pl` ignores `@disable_cascades`.

Reply:
I guess @levitte is working on a fix, see #8897

Reply:
Thanks @levitte for working on this (general) issue and @mspncp pointing us to that activity.

@bernd-edlinger, the revised solution in #8897 will solve the issue you mentioned above.
As soon as that PR has been merged we can take it over here by rebasing this chunk 3 on it.
So in my view there is no need for this technical issue to hold up your further review of chunk 3.

Reply:
except this issue, this part looks good.

Reply:
Pleased to hear.